データベース設計

Ryosuker opened the pull request:
# What
データベースの設計

# Why
フリーマーケットサイトを再現するために必要な、データベースの設計を行う

ER図：https://gyazo.com/8eaeec08921a87bcd05a0c8e535d26c3

Ryosuker committed:
データベース設計

Ryosuker committed:
カラム型変更

kota0901 commented on README.md:
```diff
@@ -1,24 +1,60 @@
+| email              | string | null: false, unique: true |
+| encrypted_password | string | null: false               |
+| nickname           | string | null: false               |
+| first_name         | string | null: false               |
+| last_name          | string | null: false               |
```
今回開発するフリマはカタカナの名字名前も保存するので、名字名前のそれぞれカタカナ用のカラムを設けましょう。

kota0901 commented on README.md:
```diff
@@ -1,24 +1,60 @@
+| item_name          | string     | null: false                    |
+| price              | integer    | null: false                    |
+| description        | text       | null: false                    |
+| category           | string     | null: false                    |
+| condition          | string     | null: false                    |
+| shipping_fee_payer | string     | null: false                    |
+| shipping_area      | string     | null: false                    |
+| shipping_days      | string     | null: false                    |
```
**商品の状態, 配送料の負担, 発送元の地域,発送までの日数,カテゴリーのカラム**

上記カラムはactivehashを用いた実装になるため、activehashのルールに沿ったカラム名と型にしましょう。
activehashにつきましては以下カリキュラムをご参考にいただければと思います。

Lesson1 最終課題補足カリキュラム
カテゴリーの選択を工夫しよう

kota0901 commented on README.md:
```diff
@@ -1,24 +1,60 @@
+| prefectures   | string     | null: false                    |
+| city          | string     | null: false                    |
+| street_line   | string     | null: false                    |
+| building_name | string     | null: false                    |
```
建物名のカラムは登録が任意のため、null: false（NOT NULL制約） は不要となります。
よって削除しましょう。

参考カリキュラム
```ruby
フェーズ：応用
タイトル：サービス設計〜設計をしよう〜
目次：制約を学ぼう
```

kota0901 reviewed (commented):
レビュー依頼ありがとうございます！
修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```

Ryosuker committed:
修正

-Userモデルのカナ列の追加
-Itemモデルのアクティブハッシュ用いた列に修正
-Addressesモデルの建物名のカラムオプション修正

minachanchan reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

コメントいたしましたので、ご確認頂けますと幸いでございます。

指摘内容の修正がお済みになりましたら、再度レビュー依頼くださいませ。
その際、ER図もREADMEの内容と合わせて修正とお願いいたします。
**次回のレビュー依頼の際に最新のER図のURLをまたこちらのプルリクエストに添付頂ければと存じます。**

実装完了まで引き続き、一緒に頑張りましょう！
何卒よろしくお願いいたします。

Ryosuker commented:
ER図を最新のものに更新しました。

Ryosuker committed:
修正２

-itemモデル内アクティブハッシュに関連するカラムは"_id"に命名
-purchaseモデル内、addressカラムを削除、アソシエーション変更
-addressモデル、postal_code、phone_numberをstring型に変更

Akinori0123 reviewed (commented):
レビュー依頼いただきありがとうございます！
コミット内容確認いたしました。コメントいたしましたのでご確認の程よろしくお願いいたします。

Ryosuker committed:
修正３

Akinori0123 reviewed (commented):
レビュー依頼いただきありがとうございます！
前回の修正バッチリですね。LGTMです。
マージ後、mainブランチでpullを行なっていただければと思います。

引き続き学習頑張っていきましょう！
